src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java

```diff
@@ -0,0 +1,20 @@
+package com.example.onlinebookstore.dto;
+
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```

Comment:
Id is assigned by DB, when you send request to create something you should not pass Id. User will obtain Id in response after creation
```suggestion
```